### ISSUE TITLE
Add active failure recovery option to reduce app failures

### ISSUE THREAD

**Issue by aronica** [PhantomThief/thrift-pool-client#2]
FailoverCheckingStrategy currently relies on cache expire of google guava, the thrift server may still be abnormal after the cache is expired, next time the application request will failed.

**Comment by PhantomThief** [comment]
You're right. The current strategy is a very simple one without any active checking strategy.
I've made an independent failover strategy here:
https://github.com/PhantomThief/simple-failover-java

and yet to be ported to this project.
